structuralcodes/sections/_rectangularRC.py

```diff
@@ -0,0 +1,580 @@
+
+                     z ▲
+                       │
+                       │
+            ┌──────────┼───────────┐
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          └───────────┼─────►
+            │                      │     y
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            └──────────────────────┘
+
+            RectangularRCSection.fromGeometry(Geometry(itPolygons,itMaterials))
+
+
+    """
+
+    def __init__(
+        self,
+        width: float,
+        height: float,
+        cover: float,
+        reinforcement_data: ReinforcementData,
+        concrete: Concrete,
+        reinforcement: ElasticPlastic,
+        confined_concrete: t.Union[Concrete, str, None] = None,
+        stirrups_data: t.Optional[StirrupsData] = None,
+        name: t.Optional[str] = None,
+    ) -> None:
+        if name is None:
+            name = 'RectangularRCSection'
+        super().__init__(name)
+
+        # section geometry
+        self.height = height
+        self.width = width
+        self.cover = cover
+        self.geometry: SurfaceGeometry
+        self.create_geometry()
+
+        # materials
+        self.concrete: Material = concrete
+        self.reinforcement: ElasticPlastic = reinforcement
+        if confined_concrete is not None:
+            if isinstance(confined_concrete, str):
+                # provided a string, check if it auto
+                if (
+                    confined_concrete.lower() == 'automatic'
+                    or confined_concrete.lower() == 'auto'
+                ):
+                    # TODO: compute automatic confinement
+                    if stirrups_data is None:
+                        raise ValueError(
+                            'Asked for automatic confinement \
+                            computation, but stirrups_data is None \
+                            \nPlease provide a proper stirrups_data'
+                        )
+                    self.confined_concrete = concrete
+                else:
+                    raise ValueError(f'Value "{confined_concrete}" unknown')
+            elif isinstance(confined_concrete, Concrete):
+                # provided a Concrete material, used that one
+                self.confined_concrete = confined_concrete
+        else:
+            self.confined_concrete = concrete
+
+        # stirrups
+        if stirrups_data is None:
+            # If no stirrups are provided by default it is assumed the
+            # diameter is 8 mm - The spacing is not significant
+            stirrups_data = StirrupsData(8, 150)
```

Comment:
@talledodiego I agree that the diameter of the stirrups diameter is a needed parameter for the cross section, but the spacing seems misplaced in my opinion. I view the spacing of the reinforcement to be a parameter of the structure (i.e. beam or column). It could be either, it depends on the object hierarchy that is designed for this code.